doc/source/dev/core-dev/deprecations.rst.inc

```diff
@@ -7,25 +7,47 @@ There are various reasons for wanting to remove existing functionality: it's
 buggy, the API isn't understandable, it's superseded by functionality with
 better performance, it needs to be moved to another SciPy submodule, etc.
 
-In general it's not a good idea to remove something without warning users about
-that removal first.  Therefore this is what should be done before removing
+In general, it's not a good idea to remove something or make other backwards
+compatibility breaking changes without warning users about
+the change first. Therefore this is what should be done before deprecating
 something from the public API:
 
 #. Propose to deprecate the functionality on the scipy-dev mailing list and get
    agreement that that's OK.
-#. Add a ``DeprecationWarning`` for it, which states that the functionality was
-   deprecated, and in which release. For Cython APIs, see
+#. When changing a function signature (e.g. deprecating a keyword argument), this
+   opportunity should be used to convert that function to keyword-only usage,
+   see `GH-14714<https://github.com/scipy/scipy/issues/14714>`_.
+#. Add a ``DeprecationWarning`` that is triggered by using the deprecated API.
+   The ``stacklevel`` must be set correctly so that the warning refers
+   to the function’s caller, rather than the ``warnings.warn()`` call itself.
+   For Cython APIs, see
    :ref:`deprecating-public-cython-api` for the practical steps.
-#. Mention the deprecation in the release notes for that release.
-#. Wait till at least 6 months after the release date of the release that
+#. For behavior changes (rather than outright removals), it must be possible for
+   the user to silence the warning without doing manual filtering or warnings.
+   In other words, a replacement API needs to be ready, and the warning should
+   contain clear instructions what needs to be done to avoid the warning.
```

Comment:
Typo snuck in (my fault)
```suggestion
#. For behavior changes (rather than outright removals), it must be possible for
   the user to silence the warning without doing manual filtering of warnings.
   In other words, a replacement API needs to be ready, and the warning should
   contain clear instructions what needs to be done to avoid the warning.
```